### PR TITLE
Release pyDARNio v1.3

### PR DESCRIPTION
**Pull request by RemingtonRohel**
# Scope 

Support of Borealis v0.7 files and `snd` files, removal of `deepdish` dependency.

## Changes

* `deepdish` library no longer maintained, introducing breaking changes with new numpy version. This dependency has been removed.
* `snd` files for sounding data are now supported.
* Borealis v0.7 files are now supported.
* Bumps Python version requirement to Python 3.8+

## Approval

**Number of approvals:** 3

## Testing

- [ ] Test I/O of Borealis v0.6 and v0.7 files
- [ ] Test I/O of `snd` files
- [ ] Test installation with newer numpy versions (e.g. v2.0.0) and verify that importing pydarnio works

### CODE REVIEW

**Comment by egthomas** [comment]
@RemingtonRohel thanks for opening this, it will be nice to get the `snd` capabilities out there.  Is it possible to add the `SND` format type to the list of supported DMAP files on the main documentation page?

https://github.com/SuperDARN/pyDARNio/blob/release/v1.3/docs/index.md
https://pydarnio.readthedocs.io/en/latest/

**Comment by RemingtonRohel** [comment]
Absolutely, thanks for the suggestion, I would have completely missed doing that!

File: pydarnio/borealis/borealis_formats.py [hunk 1]
```diff
@@ -55,6 +59,537 @@
...
+        Get the mapping of Borealis array data fields to its type
+
+        Returns
+        -------
+        A dictionary containing data fields as keys and the data field variable
+        types as values.
+        """
+        return {
+            # States what order the data is in. Describes the data layout.
+            "antenna_arrays_order": np.unicode_,
```
Comment (carleyjmartin) [inline pydarnio/borealis/borealis_formats.py:206]: When testing with the newest numpy version I'm getting an error for this line using unicode_


```
Traceback (most recent call last):
  File "/Users/carley/Documents/pydarnio/test_borealis.py", line 6, in <module>
    borealis_reader = pydarnio.BorealisRead(bfiq_site_filename, 'rawacf', 'site')
  File "/Users/carley/Documents/testing/venv-testpydarnio_PR_72/lib/python3.10/site-packages/pydarnio/borealis/borealis.py", line 136, in __init__
    self._reader = BorealisSiteRead(self.filename,
  File "/Users/carley/Documents/testing/venv-testpydarnio_PR_72/lib/python3.10/site-packages/pydarnio/borealis/borealis_site.py", line 149, in __init__
    self.read_file()
  File "/Users/carley/Documents/testing/venv-testpydarnio_PR_72/lib/python3.10/site-packages/pydarnio/borealis/borealis_site.py", line 267, in read_file
    dataset_types = self.format.site_array_dtypes()
  File "/Users/carley/Documents/testing/venv-testpydarnio_PR_72/lib/python3.10/site-packages/pydarnio/borealis/base_format.py", line 833, in site_array_dtypes
    cls.site_array_dtypes_fields()}
  File "/Users/carley/Documents/testing/venv-testpydarnio_PR_72/lib/python3.10/site-packages/pydarnio/borealis/base_format.py", line 813, in site_array_dtypes_fields
    return [k for k in cls.site_fields() if k in
  File "/Users/carley/Documents/testing/venv-testpydarnio_PR_72/lib/python3.10/site-packages/pydarnio/borealis/base_format.py", line 814, in <listcomp>
    list(cls.array_dtypes().keys())]
  File "/Users/carley/Documents/testing/venv-testpydarnio_PR_72/lib/python3.10/site-packages/pydarnio/borealis/borealis_formats.py", line 807, in array_dtypes
    return cls.fields.array_types('rawacf')
  File "/Users/carley/Documents/testing/venv-testpydarnio_PR_72/lib/python3.10/site-packages/pydarnio/borealis/borealis_formats.py", line 287, in array_types
    array_elements = cls.all_array_types()
  File "/Users/carley/Documents/testing/venv-testpydarnio_PR_72/lib/python3.10/site-packages/pydarnio/borealis/borealis_formats.py", line 206, in all_array_types
    "antenna_arrays_order": np.unicode_,
  File "/Users/carley/Documents/testing/venv-testpydarnio_PR_72/lib/python3.10/site-packages/numpy/__init__.py", line 397, in __getattr__
    raise AttributeError(
AttributeError: `np.unicode_` was removed in the NumPy 2.0 release. Use `np.str_` instead.
```

**Review by carleyjmartin — CHANGES_REQUESTED** [review]
Installed via
`pip3 install git+https://github.com/superdarn/pydarnio@release/v1.3`
successfully.

Plotted a few things with the latest release of pyDARN with no issues reading data. Untested with develop as we should fix on pyDARN end if there are issues. 

Original testing with snd files still works fine and reads data.

There was a single issue with. borealis_formats.py and the new Numpy version - see comment.

**Review by carleyjmartin — APPROVED** [review]

**Comment by tjk584** [comment]
## Tested with Inuvik dataflow
I've pulled this branch at Inuvik to run the dataflow. Following the changes in commits https://github.com/SuperDARN/pyDARNio/pull/72/commits/d82073bd803dd0665989a5e29199bf399ea61de0 and https://github.com/SuperDARN/pyDARNio/pull/72/commits/318c2ba5feffe412ed2b76f7ce3cf2736d82ae39, the data flow is successfully restructuring files as expected.

Files tested on: 
- v0.7 rawacf hdf5.site 
- v0.7 antennas_iq hdf5.site

Operation tested:
- Restructuring `.hdf5.site` record files to `.hdf5` array files

Script used in testing: https://github.com/SuperDARNCanada/data_flow/blob/main/site-linux/convert_and_restructure

**Pull request merged**